app-catalog/src/components/charts/List.tsx

```diff
@@ -39,53 +46,57 @@ export function ChartsList({ fetchCharts = fetchChartsFromArtifact }) {
     { title: 'Storage', value: 7 },
     { title: 'Streaming and messaging', value: 8 },
   ];
-  const [charts, setCharts] = useState<any[] | null>(null);
+  const [charts, setCharts] = useState<any | null>(null);
   const [openEditor, setEditorOpen] = useState<boolean>(false);
   const [page, setPage] = useState(1);
   const [chartsTotalCount, setChartsTotalCount] = useState(0);
   const [chartCategory, setChartCategory] = useState(helmChartCategoryList[0]);
   const [search, setSearch] = useState('');
   const [selectedChartForInstall, setSelectedChartForInstall] = useState<any | null>(null);
 
+  // note: since we default to true for showVerified and the settings page is not accessible from anywhere else but the list comp
+  // we must have the default value here and have it imported for use in the settings tab
+  const config = store.get();
+  const [currentConfig, setCurrentConfig] = useState(config);
+
   useEffect(() => {
-    setCharts(null);
-    fetchCharts(search, chartCategory, page).then(response => {
-      setCharts(response.packages);
-      const facets = response.facets;
-      const categoryOptions = facets.find(
-        (facet: {
-          title: string;
-          options: {
-            name: string;
-            total: number;
-          }[];
-        }) => facet.title === 'Category'
-      ).options;
-      if (chartCategory.title === 'All') {
-        const totalCount = categoryOptions.reduce(
-          (
-            acc: number,
-            option: {
-              name: string;
-              total: number;
-            }
-          ) => acc + option.total,
-          0
-        );
-        setChartsTotalCount(totalCount);
-        return;
-      }
-      const totalCount = categoryOptions.find(
-        (option: { name: string; total: number }) => option.name === chartCategory?.title
-      ).total;
-      setChartsTotalCount(totalCount);
-    });
-  }, [page, chartCategory, search]);
+    let updatedConfig;
+    if (config && config.showVerified !== undefined) {
+      console.log('config.showVerified', config.showVerified);
+      updatedConfig = { ...config };
+    } else {
+      updatedConfig = { ...config, showVerified: true };
+    }
+    store.set(updatedConfig);
+    setCurrentConfig(store.get());
+  }, []);
 
+  // note: This useEffect goes first to prevent error render when changing catagetory from the last page of any view
   useEffect(() => {
     setPage(1);
   }, [chartCategory, search]);
 
+  useEffect(() => {
+    setCharts(null);
+    const showVerified = currentConfig?.showVerified;
+
+    async function fetchData() {
+      try {
+        const response: any = await fetchCharts(search, showVerified, chartCategory, page);
+        const packages = response.dataResponse.packages;
+        const total = parseInt(response.total);
+        setCharts(packages);
+        // note: the total number of charts is located by the helm chart category with the id of 0 within "kind" facet
```

Comment:
I don't understand this comment at all, what does it mean?

Reply:
this is an old note